plugin/storage/memory/factory_test.go

```diff
@@ -76,15 +76,8 @@ func TestPublishOpts(t *testing.T) {
 	command.ParseFlags([]string{"--memory.max-traces=100"})
 	f.InitFromViper(v, zap.NewNop())
 
-	baseMetrics := metricstest.NewFactory(time.Second)
-	defer baseMetrics.Stop()
-	forkFactory := metricstest.NewFactory(time.Second)
-	defer forkFactory.Stop()
-	metricsFactory := fork.New("internal", forkFactory, baseMetrics)
+	metricsFactory := metricstest.NewFactory(time.Second)
```

Comment:
you can just use metrics.NullFactory, since you're not validating anything with it